### PR TITLE
Lf menu

### PR DESCRIPTION
**Pull request by lukefrancis**
Added a menu, colors, and a material design theme as a starting point for the redesign.

Pull also includes updates to libraries to latest versions.

Will attempt to refactor SelectBuyFlowActivity to a more general template, but wanted to push this up as a starting point for review.

### CODE REVIEW

File: app/src/main/java/com/google/android/gms/samples/wallet/SelectBuyFlowActivity.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        switch (id) {
+            case android.R.id.home:
+                mDrawerLayout.openDrawer(GravityCompat.START);
+                return super.onOptionsItemSelected(item);
+            case R.id.action_settings:
+                return true;
+            case R.id.login:
+                Intent loginIntent = new Intent(this, LoginActivity.class);
+                loginIntent.putExtra(LoginActivity.EXTRA_ACTION, LoginActivity.Action.LOGIN);
+                startActivityForResult(loginIntent, REQUEST_USER_LOGIN);
```
Comment (samtstern) [inline app/src/main/java/com/google/android/gms/samples/wallet/SelectBuyFlowActivity.java:97]: nit: why start for result but not implement `onActivityResult`?

File: app/src/main/java/com/google/android/gms/samples/wallet/SelectBuyFlowActivity.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        navigationView.setNavigationItemSelectedListener(
+        new NavigationView.OnNavigationItemSelectedListener() {
+            @Override
+            public boolean onNavigationItemSelected(MenuItem menuItem) {
+                menuItem.setChecked(true);
+                if(menuItem.getItemId() == R.id.menu_option_two_step_buyflow){
+                    Intent intent = new Intent(context, ItemListActivity.class);
+                    startActivity(intent);
+                }
+                // TODO: handle navigation to one step and product screen buyflows
```
Comment (samtstern) [inline app/src/main/java/com/google/android/gms/samples/wallet/SelectBuyFlowActivity.java:65]: If this is a developer TODO might want to do something like `Toast.makeText(this, "TODO: Implement", Toast.LENGTH_SHORT).show()`

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -16,62 +16,75 @@
...
         android:targetSdkVersion="19" />
-    <uses-permission android:name="android.permission.INTERNET"/>
-    <uses-permission android:name="android.permission.GET_ACCOUNTS"/>
-    <uses-permission android:name="android.permission.USE_CREDENTIALS"/>
+
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.GET_ACCOUNTS" />
+    <uses-permission android:name="android.permission.USE_CREDENTIALS" />
+
+    <android:uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Comment (samtstern) [inline app/src/main/AndroidManifest.xml:29]: What are these permissions used for?  I don't see any code to actually grant them.

**Review by samtstern — COMMENTED** [review]
Pretty much LGTM, a few questions

File: app/src/main/java/com/google/android/gms/samples/wallet/SelectBuyFlowActivity.java (hunk 2 above)
Comment (lukefrancis) [reply app/src/main/java/com/google/android/gms/samples/wallet/SelectBuyFlowActivity.java:65]: Ah, this is a Luke TODO.  But, when I have dev TODO's later on, I'll use this.

File: app/src/main/java/com/google/android/gms/samples/wallet/SelectBuyFlowActivity.java (hunk 1 above)
Comment (lukefrancis) [reply app/src/main/java/com/google/android/gms/samples/wallet/SelectBuyFlowActivity.java:97]: Good point; I copied over code from BikestoreFragmentActivity, but neglected to pull the relevant onActivityResult code.  I'll fix that in the next commit.

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (lukefrancis) [reply app/src/main/AndroidManifest.xml:29]: Good catch - not sure how those extra permissions crept in there.

**Comment by samtstern** [comment]
@lukefrancis any reason this is closed now?

**Comment by lukefrancis** [comment]
@samtstern The pull request would have added the changes to Master, but we don't want to do that, since that would create an app breaking change.

I'll keep the lf-menu branch as a feature branch, and create pull requests for it.

**Pull request closed without merging**